FinanceManager.UnitOfWork/EntityFramework/UnitOfWorkExecuter.cs

```diff
@@ -0,0 +1,95 @@
+﻿using FinanceManager.UnitOfWork.Abstructs;
+using FinanceManager.UnitOfWork.EntityFramework.Abstracts;
+using Microsoft.EntityFrameworkCore;
+
+namespace FinanceManager.UnitOfWork.EntityFramework;
+
+public class UnitOfWorkExecuter : IUnitOfWorkExecuter
+{
+    private readonly IUnitOfWorkFactory _unitOfWorkFactory;
+    private readonly DbContext _dbContext;
+
+    public UnitOfWorkExecuter(
+        IUnitOfWorkFactory unitOfWorkFactory,
+        DbContext dbContext)
+    {
+        _unitOfWorkFactory = unitOfWorkFactory;
+        _dbContext = dbContext;
+    }
+
+    public TResult Execute<TResult>(Func<IUnitOfWork, TResult> callRepositoryAction)
+    {
+        using var unitOfWork = _unitOfWorkFactory.CreateUnitOfWork(_dbContext);
+        try
+        {
+            var result = callRepositoryAction(unitOfWork);
+            unitOfWork.Commit();
```

Comment:
Commit may fail with very specific error and so it also would fail on rollback below, effectively hiding commit exception completely (shadow exception).